File: app/lib/data/models/order.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'order.freezed.dart';
part 'order.g.dart';

enum OrderStatus {
  @JsonValue('pending')
  pending,
  @JsonValue('accepted')
  accepted,
  @JsonValue('rider_requested')
  riderRequested,
  @JsonValue('rider_assigned')
  riderAssigned,
  @JsonValue('picked_up')
  pickedUp,
  @JsonValue('delivered')
  delivered,
  @JsonValue('ready_for_pickup')
  readyForPickup,
  @JsonValue('completed')
  completed,
  @JsonValue('cancelled')
  cancelled,
}

enum PaymentMethod {
  @JsonValue('cash')
  cash,
  @JsonValue('wallet')
  wallet,
}

@freezed
class Order with _$Order {
  const factory Order({
    required String id,
    required String buyerId,
    required String sellerId,
    required String addressId,
    required OrderStatus status,
    @Default(false) bool deliveryRequired,
    @Default(PaymentMethod.cash) PaymentMethod paymentMethod,
    required int totalAmount,
    required DateTime createdAt,
  }) = _Order;

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
}

@freezed
class OrderItemCreate with _$OrderItemCreate {
  const factory OrderItemCreate({
    required String menuItemId,
    required int qty,
    required int unitPrice,
  }) = _OrderItemCreate;

  factory OrderItemCreate.fromJson(Map<String, dynamic> json) =>
      _$OrderItemCreateFromJson(json);
}

@freezed
class OrderCreate with _$OrderCreate {
  const factory OrderCreate({
    required String sellerId,
    required String addressId,
    @Default(false) bool deliveryRequired,
    @Default(PaymentMethod.cash) PaymentMethod paymentMethod,
    required List<OrderItemCreate> items,
  }) = _OrderCreate;

  factory OrderCreate.fromJson(Map<String, dynamic> json) =>
      _$OrderCreateFromJson(json);
}

File: server/app/schemas/order.py
from enum import Enum
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel
from typing import Optional


class OrderStatus(str, Enum):
    PENDING = "pending"
    ACCEPTED = "accepted"
    RIDER_REQUESTED = "rider_requested"
    RIDER_ASSIGNED = "rider_assigned"
    PICKED_UP = "picked_up"
    DELIVERED = "delivered"
    READY_FOR_PICKUP = "ready_for_pickup"
    COMPLETED = "completed"
    CANCELLED = "cancelled"


class PaymentMethod(str, Enum):
    CASH = "cash"
    WALLET = "wallet"


class OrderItemCreate(BaseModel):
    menu_item_id: UUID
    qty: int
    unit_price: int


class OrderCreate(BaseModel):
    seller_id: UUID
    address_id: UUID
    delivery_required: bool = False
    payment_method: PaymentMethod = PaymentMethod.CASH
    items: list[OrderItemCreate]


class OrderItemResponse(BaseModel):
    id: UUID
    order_id: UUID
    menu_item_id: UUID
    qty: int
    unit_price: int
    
    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: UUID
    buyer_id: UUID
    seller_id: UUID
    address_id: UUID
    status: str
    delivery_required: bool
    payment_method: str
    total_amount: int
    created_at: datetime
    
    model_config = {"from_attributes": True}

